### PR TITLE
docs: Add JetBrains and VSCode authoring tips

### PR DESCRIPTION
**Pull request by oguzhand95**
Signed-off-by: Oğuzhan Durgun <oguzhandurgun95@gmail.com>

#### Description

This PR adds JetBrains and VSCode authoring tips to the documentation.

### CODE REVIEW

File: docs/modules/policies/pages/authoring_tips.adoc [hunk 1]
```diff
@@ -79,3 +79,44 @@ use({
     end,
 })
 ----
+
+==== JetBrains IDEs
+
+Go into the Preferences of JetBrains IDE of choice and navigate to the `JSON Schema Mappings` menu.
+
+----
+Preferences > Languages & Frameworks > Schemas and DTDs > JSON Schema Mappings
```
Comment (charithe) [inline docs/modules/policies/pages/authoring_tips.adoc:88]: I think you can use the Menu macro here: https://docs.asciidoctor.org/asciidoc/latest/macros/ui-macros/#menu-macro-syntax

File: docs/modules/policies/pages/authoring_tips.adoc [hunk 2]
```diff
@@ -79,3 +79,44 @@ use({
...
+[subs="attributes"]
+----
+Name: Cerbos
+Schema file or URL: {latest-schema-url}
+Schema version: JSON Schema Version 7
+File path pattern: cerbos/*
+----
+image:jetbrains-menu.png[alt="JetBrains JSON Schema Mappings menu",role="center-img"]
+
+NOTE: In the example above, the schema is applied to all files in the /cerbos directory.
```
Comment (charithe) [inline docs/modules/policies/pages/authoring_tips.adoc:102]: I am probably wrong here, but, Isn't the file path pattern relative in the above example? So presumably it applies to any files contained in a directory named cerbos?

```suggestion
NOTE: In the example above, the schema is applied to all files in the cerbos directory.
```

File: docs/modules/policies/pages/authoring_tips.adoc [hunk 3]
```diff
@@ -79,3 +79,44 @@ use({
...
+
+==== Visual Studio Code
+
+Before starting, make sure you know how to modify Visual Studio Code settings by taking a look at the link:https://code.visualstudio.com/docs/getstarted/settings[VSCode documentation].
+
+For Visual Studio Code, an extension is required to work with the schema.
+Please install the extension from link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml[here]
+
+After the required extension is ready, please navigate to the Visual Studio Code's `settings.json` and add the following part.
```
Comment (charithe) [inline docs/modules/policies/pages/authoring_tips.adoc:111]: Documentation doesn't usually contain words like "please" :)
 
```suggestion
If you are new to Visual Studio Code, refer to the
 link:https://code.visualstudio.com/docs/getstarted/settings[documentation] for more information about how to change settings.

Install the YAML language server extension from https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml

After the extension is installed, hit kbd:[Ctrl+,] or menu:File[Preferences > Settings] to edit settings. Expand menu:Extensions[YAML], click `Edit in settings.json` under `Yaml: Schemas`. and add the following snippet:

```

File: docs/modules/policies/pages/authoring_tips.adoc [hunk 4]
```diff
@@ -79,3 +79,44 @@ use({
...
+[source,json,subs="attributes"]
+----
+{
+  "yaml.schemas": {
+    "{latest-schema-url}": "cerbos/*"
+  }
+}
+----
+
+NOTE: In the example above, the schema is applied to all files in the /cerbos directory.
```
Comment (charithe) [inline docs/modules/policies/pages/authoring_tips.adoc:122]:
```suggestion
NOTE: In the example above, the schema is applied to all files in the cerbos directory.
```

File: docs/modules/policies/pages/authoring_tips.adoc (hunk 2 above)
Comment (oguzhand95) [reply docs/modules/policies/pages/authoring_tips.adoc:102]: Yes, you are right. My mistake.

**Pull request merged**